### ISSUE TITLE
Exception has occurred. _AssertionError ('package:flutter/src/services/hardware_keyboard.dart')

### ISSUE THREAD

**Issue by adityapandeyz** [flutter/flutter#119141]
Exception has occurred.
_AssertionError ('package:flutter/src/services/hardware_keyboard.dart': Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)': A KeyUpEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#78615(physicalKey: PhysicalKeyboardKey#7002a(usbHidUsage: "0x0007002a", debugName: "Backspace"), logicalKey: LogicalKeyboardKey#00008(keyId: "0x100000008", keyLabel: "Backspace", debugName: "Backspace"), character: null, timeStamp: 0:00:05.647760, synthesized))
![Screenshot 2023-01-25 203311](https://user-images.githubusercontent.com/40023090/214598148-9530125b-81c0-466b-899b-328e7c39ef68.png)

**Comment by darshankawar** [comment]
@adityapandeyz 
Can you provide `flutter doctor -v` along with steps to replicate and minimal code sample ?

**Comment by adityapandeyz** [comment]
`flutter doctor -v`


[√] Flutter (Channel stable, 3.7.0, on Microsoft Windows [Version 10.0.22621.1105], locale en-IN)
    • Flutter version 3.7.0 on channel stable at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (4 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at C:\Users\adity\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.1
    • Java binary at: C:\Program Files\Java\jdk-17.0.5\bin\java
    • Java version Java(TM) SE Runtime Environment (build 17.0.5+9-LTS-191)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.1)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33110.190
    • Windows 10 SDK version 10.0.22000.0

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.74.3)
    • VS Code at C:\Users\adity\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.58.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.1105]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 109.0.5414.120
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.76

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

**Comment by adityapandeyz** [comment]
**Minimal code:**]

TextField(
  textAlign: TextAlign.center,
  keyboardType: TextInputType.emailAddress,
  onChanged: (value) {
    email = value;
  },
  decoration: kTextFieldDecoration.copyWith(
    hintText: 'Enter your email',
  ),
),

const kTextFieldDecoration = InputDecoration(
  contentPadding: EdgeInsets.symmetric(
    vertical: 10.0,
    horizontal: 20.0,
  ),
  border: OutlineInputBorder(
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
  enabledBorder: OutlineInputBorder(
    borderSide: BorderSide(
      color: Colors.lightBlueAccent,
      width: 1.0,
    ),
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
  focusedBorder: OutlineInputBorder(
    borderSide: BorderSide(
      color: Colors.lightBlueAccent,
      width: 2.0,
    ),
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
);

**Comment by adityapandeyz** [comment]
```
TextField(
  textAlign: TextAlign.center,
  keyboardType: TextInputType.emailAddress,
  onChanged: (value) {
    email = value;
  },
  decoration: kTextFieldDecoration.copyWith(
    hintText: 'Enter your email',
  ),
),
```

```
const kTextFieldDecoration = InputDecoration(
  contentPadding: EdgeInsets.symmetric(
    vertical: 10.0,
    horizontal: 20.0,
  ),
  border: OutlineInputBorder(
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
  enabledBorder: OutlineInputBorder(
    borderSide: BorderSide(
      color: Colors.lightBlueAccent,
      width: 1.0,
    ),
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
  focusedBorder: OutlineInputBorder(
    borderSide: BorderSide(
      color: Colors.lightBlueAccent,
      width: 2.0,
    ),
    borderRadius: BorderRadius.all(
      Radius.circular(32.0),
    ),
  ),
);
```

**Comment by darshankawar** [comment]
Thanks for the update @adityapandeyz 
Can you try the same using latest master as well and see if the behavior persists on it too ?

`flutter channel master`
`flutter upgrade`

**Comment by adityapandeyz** [comment]
Problem still persists after upgrading to Flutter master channel

**Comment by darshankawar** [comment]
@adityapandeyz 
What are the steps to replicate this error ? Are you using a physical hardware keyboard to input the text ? or using laptop ? After inputing a text in the field, are you using backspace to clear the text that leads to this error ?

**Comment by GusRodrigues86** [comment]
This happens with me fairly regularly on all projects.

What causes? Just using my macbook keyboard. at random times it will fails and remain on that state until I stop the project and rebuild it.

**Comment by adityapandeyz** [comment]
Yes, I was using a laptop to input text, and after inputting text, I used the backspace key to clear a mistake in the text field that eventually led to this error.

**Comment by darshankawar** [comment]
I tried the below runnable code sample and ran it on macOS desktop, but was unable to replicate the error. Maybe it is related to windows + desktop setup.

<details>
<summary> runnable code sample </summary>

```

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  var kTextFieldDecoration = InputDecoration(
    contentPadding: EdgeInsets.symmetric(
      vertical: 10.0,
      horizontal: 20.0,
    ),
    border: OutlineInputBorder(
      borderRadius: BorderRadius.all(
        Radius.circular(32.0),
      ),
    ),
    enabledBorder: OutlineInputBorder(
      borderSide: BorderSide(
        color: Colors.lightBlueAccent,
        width: 1.0,
      ),
      borderRadius: BorderRadius.all(
        Radius.circular(32.0),
      ),
    ),
    focusedBorder: OutlineInputBorder(
      borderSide: BorderSide(
        color: Colors.lightBlueAccent,
        width: 2.0,
      ),
      borderRadius: BorderRadius.all(
        Radius.circular(32.0),
      ),
    ),
  );

  late String email;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Scaffold(
        body: Center(
          child: TextField(
            textAlign: TextAlign.center,
            keyboardType: TextInputType.emailAddress,
            onChanged: (value) {
              email = value;
            },
            decoration: kTextFieldDecoration.copyWith(
              hintText: 'Enter your email',
            ),
          ),
        ),
      ),
    );
  }
}

```

</details>


There are few similar issues that have been reported already and could be related with same root-cause:

https://github.com/flutter/flutter/issues/99653
https://github.com/flutter/flutter/issues/116953

So I suggest you to follow-up in above issues and you may also share your findings in it for better tracking.
Closing this as a duplicate. If you disagree, write in comments and I'll reopen it.
